dlt/extract/utils.py

```diff
@@ -180,39 +180,20 @@ def wrap_async_iterator(
     gen: AsyncIterator[TDataItems],
 ) -> Generator[Awaitable[TDataItems], None, None]:
     """Wraps an async generator into a list of awaitables"""
-    exhausted = False
-    busy = False
-
-    # creates an awaitable that will return the next item from the async generator
-    async def run() -> TDataItems:
-        nonlocal exhausted
+    loop = asyncio.get_event_loop()
+    should_stop = False
+    try:
         try:
-            # if marked exhausted by the main thread and we are wrapping a generator
-            # we can close it here
-            if exhausted:
-                raise StopAsyncIteration()
-            item = await gen.__anext__()
-            return item
-        # on stop iteration mark as exhausted
-        # also called when futures are cancelled
+            while True:
+                if should_stop:
+                    break
+                yield loop.run_until_complete(gen.__anext__())  # type: ignore[arg-type]
```

Comment:
I used eventloop because with the old implementation it has been complaining about not awaited generator
```py
tests/extract/test_sources.py::test_add_limit_async
  /[...]/dlt/dlt/extract/pipe_iterator.py:275: RuntimeWarning: coroutine 'wrap_async_iterator.<locals>.run' was never awaited
    pipe_item = next(gen)
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
```